### DISCUSSION TITLE
Changing the default output pins ?

### DISCUSSION THREAD

**Discussion by jenom1957** [ArminJo/Talkie#38] [General]
I would like to change the default output pins of 3 and 11 to something else......like 11 & 12
searched through a few files, and have no clue if it is possible or how?
Any suggestion ? Thanks !

**Comment by ArminJo** [comment]
Have you read https://github.com/ArminJo/Talkie#pin-and-timer-mapping-table-for-different-platforms ?

**Reply by jenom1957** [reply]
Yes, pin 3 & 11 for Arduino Uno, how do I change it to something else?

**Reply by madmacks59** [reply]
In the Talkie.cpp file you should find a block of code like this...

---------------------------------------------------
    /*
     * Plain ATmega e.g. 328P here
     *
     * Timer 2 set up as a 62500 Hz / 16 us 8Bit PWM for 16 MHz.
     * The PWM 'buzz' is well above human hearing range and is very easy to filter out.
     */
    TCCR2A = _BV(WGM21) | _BV(WGM20); // Fast PWM
    if (NonInvertedOutputPin) {
        NonInvertedOutputPin = 3; // OC2B
        pinMode(NonInvertedOutputPin, OUTPUT);
        TCCR2A |= _BV(COM2B1); // OC2B non-inverting mode
    }
    if (InvertedOutputPin) {
        InvertedOutputPin = 11; // OC2A
        pinMode(InvertedOutputPin, OUTPUT);
        TCCR2A |= _BV(COM2A1) | _BV(COM2A0); // OC2A inverting mode
    }
--------------------------------------------------

In that block, I believe you can change the two pin definitions to something else (3, 11). But, I haven't done a detailed search to confirm there is not some other Define block that also sets the pins. Maybe that will resolve your issue. Mod the file, save it, and recompile your .ino file and the new pins should be used.

**Comment by madmacks59** [comment]
I too would like to change the output pin. In my case I am using an audio amp that has a Left and Right channel, one connected to ESP32 pin 25 and the other to pin 26. In my code, OI'd like to dynamically change the pin from 25 to 26 for audio output, then change it from 26 back to 25. I've using the ESP32 to monitor a tank filled with water, and I have once speaker outside the tank shed and one inside the tank shed. I want to alterantly play a sound on the outside and the inside of the shed. 

It looks to me like the pin is defined during compile (using #define ESP32, which then uses a macro to define the pin as 25). What I am hoping for is a method call to the voice object that will let me change the Pin on the fly. Short of that is there a hard coded way to change the output pin?

**Comment by madmacks59** [comment]
Well, I just coded and found out...the below works just fine and allows dynamic pin changes to the running code.

voice.NonInvertedOutputPin = 25;
and
voice.NonInvertedOutputPin = 26;
